### DISCUSSION TITLE
Thanks

### DISCUSSION THREAD

**Discussion by dougle03** [Sleeper85/esphome-jk-bms-can#5] [General]
Thankyou @Sleeper85  for keeping this open unlike @UKSA007 who has decided to try and profit from other people's works. A soon as my new battery pack is up and running I'll be testing and will give feedback and assistance where I can.

**Comment by Uksa007** [comment]
You are the exact type of person that caused me to stop supporting this open source, as you:

- Complain about everything
- Want everything for free
- Don't provide any financial support
- Want many hours of support for free, to help build hardware/firmware which you have no experience with.
- Generally have no clue

If you weren't so busy complaining about everything you would see that it is me who:

- Spent 1000's of hours researching how inverter communications work.
- Spent 1000's of hours writing code and testing.
- Spent 1000's of hours designing PCB, sourcing parts
- Manufactures the Boards. 
- Provides support

If you spent five minutes you would see that Syssi code is simple serial reading
And if you bothered to check his licence allows for all kinds of use.

![image](https://github.com/Sleeper85/esphome-jk-bms-can/assets/8242576/6c38ba2a-4618-4493-81ae-085da0bf0c12)

**Reply by dougle03** [reply]
Let us be clear. What caused you to close source your product was the allure of cash substantially on the backs of others freely given hard work; nothing more, else you wouldn't be charging the sums you are for your product.. Had your sell price been reasonable I very much doubt I'd be remotely interested in this.

You state on your distro that you removed the yaml because you didn't not have the time for (free) support. And in itself this is perfectly, understandable, however you could easily have continued to publish the yaml but with the disclaimer that you would provide no end user support, something that many devs do, especially where their work is 95% derived from the open source community.
What you've actually done is to squirrel away other people's hard work, done for no payment and decided to attempt to profit from it. I don't doubt you have put some effort into your current version code, and indeed a few hours in kicad, and equally to be clear, I have no problems with you marketing a hardware solution, but when you do that using a platform (ESPHome) built entirely on the backs of OS devs like me, whilst taking a substantial portion of your code from another OS dev, then to me that's ethically a poor show, even Victron share a substantial part of their source, but not you.

If this product cost you 4000+ hours to develop, then may I humbly suggest you perhaps look for alternative means of expressing your interests. And let's not kid ourselves, pulling together an existing ESP32 reference design development board with a very well understood CAN bus chip is really not that complicated a task, especially considering all of the abstract code is already written for you, and freely available. 
I really hope your product is a success, however niche it is, I reserve the right to call you out at every opportunity including the various Victron Groups I run on Facebook, and the ESPHome group. And if there is anything that I've said that is not true, then please don't hesitate to correct me.

**Reply by Uksa007** [reply]
The cost barely cover the labour to build the product, ship it and provide support for it. 
It's not mass produced.

It easy to throw stones when all the hard work has been done for you. 
When I started this there was no ESP32 support for any CAN protocol, I had to figure it all out. 
It's multiple years of work and you are unhappy that I charge people for a hardware solution, give me a break.

Guess you just going to ignore that fact that Sysi code is a small part and is Permissively licenced.

Every single product/software is based on work by someone else, if the eg IEEE didn't make the standards no one would have standards for anything, nothing would work.

This code is still there I'm just not providing updates anymore, and it doesn't compile anymore.
If you don't support the open source version high likelihood of many unhappy people as they can't get it to work, which has a negative impact on your name and hardware product. 

Better in my option to offer just one solution, which I believe solves all the problems for everyone:
Hence the Hardware Interface kit was born.
Removed frustration for the user trying to build hardware/firmware, as many have little to no electronics and coding experience. 
The user ends up with a professional board, as opposed to hacked together modules which are unreliable at best.
I don't spend all my time trying to support people building hardware and compiling firmware, breaking the code, etc.
It is build and tested, plug and play.
Win Win.
Yes it cost money, life is not alway free.

**Reply by dougle03** [reply]
"when all the hard work has been done for you", finally the penny drops...
Had you not used the default WiFi portal and http portal then I doubt anyone would have taken any interest in it's origins. You spent 1000's of hours coding but neglected to spend 5 mins on the CSS. 
I agree, standards are important, however there is a difference between adopting a standard and grossly overpricing someone else's work, even if that's technically allowed. The hardware must be costing no more than £6-10 total, you charge extra for postage and handling so that leaves the best part of £200 for what exactly...? 
As I said, I wish you well, but I do reserve the right to call you out everywhere I see your product promoted...
The OS community is not about greed, perhaps you didn't get that memo...

**Reply by Uksa007** [reply]
The Penny drops you have no clue what you are talking about. 
The standard interface kit cost $110 USD or 86.87 Pound sterling plus postage, less if you are a Patreon member. 
Start deducting cost, Paypal fees, PCB manufacture, components, cables assemblies,  shipping to me and import taxes/duties on everything. Shipping supplies etc. 
Roughly 2hrs labour to build and test each unit then ship.
It's a labour of love.
Perhaps you didn't get the memo, not everything has to be open source.

**Reply by kommando828** [reply]
I note in your listing of costs there are no fees for the open source royalties. I have nothing against commercial operations just as long as they pay their dues to the source they used as the starting point, which you seem to have neglected to consider.

**Comment by dougle03** [comment]
@Sleeper85 If you want any assistance promoting your project here please don't hesitate to let me know. I manage both the ESPHome and Victron Energy User groups on FB and would be very happy to assist. Cheers, P

**Reply by Sleeper85** [reply]
Thank you for offering me your help. Can you contact me privately on DIY solar forum?

https://diysolarforum.com/members/sleeper85.71002/

**Reply by dougle03** [reply]
No worries.

**Reply by arzaman** [reply]
Same here ! 
thanks for your great contribution to keep open sourced the code
I have promoted your work and support for this great project in a couple of major Italian solar DIY forums, receiving lots of positive reactions!
